### PR TITLE
URL parsing fixes

### PR DESCRIPTION
**Pull request by ssanjay1**
Allow for variable entry points for urls rather than just assuming hostname will be entry point

### CODE REVIEW

File: packages/oauth/src/createConfidentialOauthClient.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function createConfidentialOauthClient(
   url: string,
   scopes: string[] = ["api:read-data", "api:write-data"],
   fetchFn: typeof globalThis.fetch = globalThis.fetch,
-  ctxPath: string = "/multipass",
+  ctxPath: string = "multipass",
```
Comment (ssanjay1) [inline packages/oauth/src/createConfidentialOauthClient.ts:42]: Allows for custom entry points in url for multipass as well

File: packages/client/src/actions/actions.test.ts [hunk 2]
```diff
@@ -124,6 +131,25 @@ describe("actions", () => {
       `);
   });
 
+  it("returns validation directly on validateOnly mode, with custom entry point in URL", async () => {
+    const result = await customEntryPointClient(moveOffice).applyAction({
```
Comment (ssanjay1) [inline packages/client/src/actions/actions.test.ts:135]: Add test to make sure URL resolution still works with custom entrypoint

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts [hunk 3]
```diff
@@ -332,8 +332,9 @@ export class ObjectSetListenerWebsocket {
     if (this.#ws == null) {
       const { baseUrl, tokenProvider } = this.#client;
       const base = new URL(baseUrl);
-      const url =
-        `wss://${base.host}/api/v2/ontologySubscriptions/ontologies/${this.#client.ontologyRid}/streamSubscriptions`;
+      const url = `wss://${base.host}${
+        base.pathname === "/" ? "" : base.pathname
```
Comment (ericanderson) [inline packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:336]: Should we be using `new URL` for this?

**Review by ericanderson — COMMENTED** [review]
Yeah we need to find a way to test all the edge cases here or we are going to end up breaking something in a future update

File: packages/client/src/createMinimalClient.ts [hunk 4]
```diff
@@ -52,10 +52,13 @@ export function createMinimalClient(
       throw new Error(`Invalid stack URL: ${baseUrl}${hint}`);
     }
   }
-
+  const processedBaseUrl = new URL(baseUrl);
+  processedBaseUrl.pathname += processedBaseUrl.pathname.endsWith("/")
```
Comment (ssanjay1) [inline packages/client/src/createMinimalClient.ts:56]: If we don't add `/` at the end of the baseUrl, we lose the last parameter in the URL when appending to the URL to hit APIs

File: packages/client/src/object/object.test.ts [hunk 5]
```diff
@@ -75,6 +82,31 @@ describe("OsdkObject", () => {
       // it should have the prototype that we assign at hydration time
       expect(Object.keys(employee.$link.lead)).toBeDefined();
     });
+    it("loads an employee with custom client", async () => {
```
Comment (ssanjay1) [inline packages/client/src/object/object.test.ts:85]: Added another test to make sure the API url appending works as intended

File: packages/oauth/src/AuthorizationServer.test.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  beforeAll,
+  describe,
+  expect,
+  expectTypeOf,
+  it,
+} from "vitest";
+
+import { createAuthorizationServer } from "./common.js";
+
+describe("Create authorization server", () => {
```
Comment (ssanjay1) [inline packages/oauth/src/AuthorizationServer.test.ts:28]: Testing that the multipass URL building for oauth clients work

File: packages/oauth/src/createPublicOauthClient.ts [hunk 7]
```diff
@@ -104,7 +104,7 @@ export function createPublicOauthClient(
  * @param {string} postLoginPage - URL to return to after completed authentication cycle (defaults to `window.location.toString()`)
  * @param {string[]} scopes - OAuth scopes to request. If not provided, defaults to `["api:read-data", "api:write-data"]`
  * @param {typeof globalThis.fetch} fetchFn - Custom fetch function to use for requests (defaults to `globalThis.fetch`)
- * @param {string} ctxPath - Context path for the authorization server (defaults to "/multipass")
+ * @param {string} ctxPath - Context path for the authorization server (defaults to "multipass")
```
Comment (ssanjay1) [inline packages/oauth/src/createPublicOauthClient.ts:107]: Just modified the JS docs here to align with reality now

File: packages/shared.test/src/handlers/actionEndpoints.ts [hunk 8]
```diff
@@ -57,6 +57,16 @@ export const actionHandlers: Array<RequestHandler> = [
...
   ),
 
+  /**
+   * Apply an Action with custom URL entrypoint
+   */
+  handleOpenApiCall(
+    OntologiesV2.Actions.apply,
+    ["ontologyApiName", "actionType"],
+    handleAction<SyncApplyActionResponseV2>,
+    "https://stack.palantirCustom.com/foo/first/someStuff/",
```
Comment (ssanjay1) [inline packages/shared.test/src/handlers/actionEndpoints.ts:67]: Added another handler for the custom URL

File: packages/shared.test/src/handlers/objectSetEndpoints.ts [hunk 9]
```diff
@@ -83,3 +72,36 @@ export const objectSetHandlers: Array<RequestHandler> = [
     },
   ),
 ];
+
+async function handleLoadObjectSet(
```
Comment (ssanjay1) [inline packages/shared.test/src/handlers/objectSetEndpoints.ts:76]: Same logic, just refactored for code re-use with the new handler

File: packages/shared.test/src/handlers/ontologyMetadataEndpoints.ts [hunk 10]
```diff
@@ -535,3 +527,41 @@ export const ontologyMetadataEndpoint: Array<RequestHandler> = [
     ),
   ),
 ];
+
+async function handleInterfaceGet(
```
Comment (ssanjay1) [inline packages/shared.test/src/handlers/ontologyMetadataEndpoints.ts:531]: Same logic as before, just refactored for new endpoint handler

File: packages/client/src/object/object.test.ts (hunk 5 above)
Comment (nihalbhatnagar) [reply packages/client/src/object/object.test.ts:84]: So we know the url appending works since we're hitting the right stub right?

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts [hunk 11]
```diff
@@ -333,8 +333,12 @@ export class ObjectSetListenerWebsocket {
     if (this.#ws == null) {
       const { baseUrl, tokenProvider } = this.#client;
       const base = new URL(baseUrl);
-      const url =
-        `wss://${base.host}/api/v2/ontologySubscriptions/ontologies/${this.#client.ontologyRid}/streamSubscriptions`;
+      const url = new URL(
+        `api/v2/ontologySubscriptions/ontologies/${this.#client.ontologyRid}/streamSubscriptions`,
+        base,
+      );
+      url.protocol = url.protocol.replace("https", "wss");
```
Comment (nihalbhatnagar) [inline packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:340]: Can you test that this works still? I'd be fine with just an e2e or even a console log of the URL but putting this in a method and adding a test may also be nice

File: packages/client/package.json [hunk 12]
```diff
@@ -35,8 +35,8 @@
     "@osdk/api": "workspace:~",
     "@osdk/client.unstable": "workspace:*",
     "@osdk/generator-converters": "workspace:*",
-    "@osdk/internal.foundry.core": "2.5.0",
-    "@osdk/internal.foundry.ontologiesv2": "2.5.0",
+    "@osdk/internal.foundry.core": "2.6.0-beta.0",
+    "@osdk/internal.foundry.ontologiesv2": "2.6.0-beta.0",
```
Comment (nihalbhatnagar) [inline packages/client/package.json:39]: We're fine with this on a beta release? I don't see why it matters so just checking

File: packages/client/src/objectSet/ObjectSetListenerWebsocket.ts (hunk 11 above)
Comment (nihalbhatnagar) [reply packages/client/src/objectSet/ObjectSetListenerWebsocket.ts:340]: TS playground looks good

Comment (ssanjay1) [reply packages/client/src/objectSet/ObjectSetListenerWebsocket.ts]: Added a helper method and test as well

File: packages/client/package.json (hunk 12 above)
Comment (ssanjay1) [reply packages/client/package.json:39]: That should be fine cuz these are beta clients anyways

Comment (ssanjay1) [reply packages/client/package.json:39]: Also going to upgrade soon

**Review by nihalbhatnagar — APPROVED** [review]
